src/messaging.rs

```diff
@@ -51,6 +51,37 @@ impl<M, N> TargetedMessage<M, N> {
     }
 }
 
+/// Result of one step of the local state machine of a distributed algorithm. Such a result should
+/// be used and never discarded by the client of the algorithm.
+pub struct Step<N, O>
```

Comment:
Should this be annotated with `#[must_use]`?
(Edit: Ah, I just saw you annotated the methods. But I think you can also annotate a `struct`; not sure exactly what difference that makes.)

Maybe it should also print an error when dropped (or even panic — at least if `#[cfg(test)]`) if the output hasn't been consumed.

Reply:
Annotating the struct with `#[must_use]` is preferable to annotating the methods. Mostly because you don't have to remember to annotate every method that returns it.

Reply:
I tried to add both the warning and the panic in the test configuration:
```
#[must_use = "The algorithm step result must be used."]
#[cfg_attr(test, forbid(unused_must_use))]
```
The first line works but second seems to have no effect. It still prints a warning in tests and doesn't stop comilation.

Reply:
Do you need to say `allow` instead of `forbid`?

Anyway, once #120 is done, this line should become unnecessary anyway.

Reply:
It's either `forbid` or `deny`: https://doc.rust-lang.org/reference/attributes.html#lint-check-attributes

So, this works if you annotate tests rather than the type: `#[deny(unused_must_use)]`. It throws the required error.

Reply:
I tried other combinations and none of them work:
* `#![cfg_attr(test, deny(unused_must_use))] in crate root,
* same in the `messaging.rs` morule,
* `deny` in `tests/network/mod.rs`.

The only attribution that works would be module-level in each test module. If `Cargo.toml` allowed to define dev attributes, that would probably suffice.

Reply:
I think if you annotate an item with something like `allow` or `forbid`, that applies to only the code in that item. So it would be an error to not use another `must_use` item inside of `Step`'s own code.

Annotating the tests accordingly would be correct, I think. (Once the tests are updated.)